### PR TITLE
RAC-398: Add controller to know number of product and product models impacted by attribute deletion

### PR DESCRIPTION
**Pull request by StevenVAIDIE**
**Description (for Contributor and Core Developer)**
In this PR: 
- New endpoint to determine the number of product and product model impacted by an attribute deletion

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | Todo
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | Todo
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: tests/back/Pim/Structure/EndToEnd/Attribute/InternalApi/ImpactedItemByAttributeDeletionEndToEnd.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    protected function getConfiguration(): Configuration
+    {
+        return $this->catalog->useTechnicalCatalog();
+    }
+
+    private function loadFixtures(): void
+    {
+            /** @var EntityBuilder $entityBuilder */
+        $entityBuilder = $this->get('akeneo_integration_tests.catalog.fixture.build_entity');
```
Comment (SamirBoulil) [inline tests/back/Pim/Structure/EndToEnd/Attribute/InternalApi/ImpactedItemByAttributeDeletionEndToEnd.php:47]: only if you want, you could initialize the entity builder in the setUp ?

**Review by SamirBoulil — APPROVED** [review]
insta gtm

**Review by ValentinMumble — APPROVED** [review]

File: tests/back/Pim/Enrichment/EndToEnd/Product/CountItemsWithAttributeValueEndToEnd.php [hunk 2]
```diff
@@ -1,24 +1,30 @@
 <?php
 
-namespace AkeneoTest\Pim\Structure\EndToEnd\Attribute\InternalApi;
+namespace AkeneoTest\Pim\Enrichement\EndToEnd\Attribute\InternalApi;
```
Comment (ValentinMumble) [inline tests/back/Pim/Enrichment/EndToEnd/Product/CountItemsWithAttributeValueEndToEnd.php:3]:
```suggestion
namespace AkeneoTest\Pim\Enrichment\EndToEnd\Attribute\InternalApi;
```

Comment (ValentinMumble) [reply tests/back/Pim/Enrichment/EndToEnd/Product/CountItemsWithAttributeValueEndToEnd.php:3]: ah oui oui oui enriche moi

Comment (StevenVAIDIE) [reply tests/back/Pim/Enrichment/EndToEnd/Product/CountItemsWithAttributeValueEndToEnd.php]: embarrassed

**Pull request merged**